File: solutions/go/pangram/1/pangram.go
package pangram

func IsPangram(input string) bool {
	const all = (1 << 26) - 1
	mask := 0

	for _, r := range input {
		// Normalize to lowercase ASCII if it's A–Z
		if r >= 'A' && r <= 'Z' {
			r += 'a' - 'A'
		}
		// Only count a–z letters
		if r >= 'a' && r <= 'z' {
			mask |= 1 << (r - 'a')
			if mask == all {
				return true
			}
		}
	}
	return mask == all
}

File: solutions/typescript/pangram/1/pangram.ts
export function isPangram(sentence: string): boolean {
  const lower = sentence.toLowerCase();

  const letters = new Set<string>();

  for (const ch of lower) {
    if (ch >= 'a' && ch <= 'z') {
      letters.add(ch);
    }
  }

  return letters.size === 26;
}
